src/main/java/com/netgrif/application/engine/configuration/security/SecurityContextFilter.java

```diff
@@ -23,7 +23,7 @@
     /**
      * Security context service for managing user tokens
      * */
-    private ISecurityContextService securityContextService;
+    private final ISecurityContextService securityContextService;
 
     public SecurityContextFilter(ISecurityContextService securityContextService) {
```

Comment:
Is this class instantiated by us or by Spring?

If the argument is passed via DI, why is there no `@Autowired` annotation?

The documentation says it might not be required since Spring 4.3 if the class has only one constructor. I personally think the annotation adds clarity. What does @mladoniczky think about this?

Reply:
This class is instantiated by Spring. The actual version of Spring Boot (2.6.x) contains Spring 5, and it suggest to use DI via constructor instead of Autowired annotation. It is because this way is more secure, because if you are using @Autowired, everyone is allowed to create a bean using the default constructor, but with this way, everyone should be aware of the other dependencies of this bean. If you use Intellij IDEA, it warns you too, that the field injection is not recommended.

Of course I aggree, that with e.g. 20 dependencies this code would look like a mess, but on the other hand, a single bean should not contain that much dependencies.

Please, for more info refer to https://reflectoring.io/constructor-injection/, https://spring.io/blog/2007/07/11/setter-injection-versus-constructor-injection-and-the-use-of-required

Reply:
I agree with using constructor DI. I would just personally prefer it to be used with the annotation, even if the annotation is optional. That is:
`@Autowired
public SecurityContextFilter(ISecurityContextService securityContextService)`

Reply:
Ah okay, excuse me, now I understand. It is not problem to add it.

Reply:
Ah sorry now I see your concern. I misread the class name and I considered that is a bean. I corrected my first reply.